### DISCUSSION TITLE
AfterAccept

### DISCUSSION THREAD

**Discussion by taehyun517** [ANLAB-KAIST/KENSv3#202] [Project Q&A 2. Transport Layer (TCP)]
In AfterAccept test, it seems so accept() before the SYN is sent. So I blocked accept() by returning nothing. However, the expected accept is 1 which is different with client_sockets. I think the if(clientfd >=0) shouldn't work so there are no elements in client_sockets.
<img width="314" alt="hi1" src="https://github.com/ANLAB-KAIST/KENSv3/assets/85110272/20af4f70-f256-4b89-aea2-4f2b3e5d0120">
<img width="451" alt="hi2" src="https://github.com/ANLAB-KAIST/KENSv3/assets/85110272/bacf4ff6-a90c-49f3-9835-bad6ad0872e6">

**Comment by Pusnow** [comment]
You have to return the blocked `accept()` call when a new connection is established.
So you have to unblock the call in the `packetArrived()` function after connection setup.
